core/domain/state_domain.py

```diff
@@ -173,6 +174,187 @@ def validate(self):
+        self.placeholder = placeholder
+        self.author_id = author_id
+        self.instructions = instructions
+        self.validate()
+
+    def to_dict(self):
+        """Returns a dict representing this Image domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of Image instance.
+        """
+        image = {}
+        image['src'] = self.src
+        image['placeholder'] = self.placeholder
+        image['author_id'] = self.author_id
+        image['instructions'] = self.instructions
+        return image
+
+    @classmethod
+    def from_dict(cls, image_info):
+        """Return an Imagedomain object from a dict.
+
+        Args:
+            image_info: dict. The dict representation of
+                Image object.
+
+        Returns:
+            Image. The corresponding ImageAssets domain
+            object.
+        """
+        src = image_info['src']
+        placeholder = image_info['placeholder']
+        author_id = image_info['author_id']
+        instructions = image_info['instructions']
+
+        return cls(src, placeholder, author_id, instructions)
+
+    def validate(self):
+        """Validates all properties of an Image.
+
+        Raises:
+            ValidationError: One or more attributes of the ImageAssets are not
+            valid.
+        """
+        # src_re = r'^[A-Za-z0-9+/]*\.((png)|(jpeg)|(gif)|(jpg))$'
+        # if not re.match(src_re, self.src):
+        #     raise utils.ValidationError(
+        #         'Invalid image_src, received %s' % self.src)
+        if not isinstance(self.src, basestring):
+            raise utils.ValidationError(
+                'Invalid image_src, received %s' % self.src)
+        if not isinstance(self.author_id, basestring):
+            raise utils.ValidationError(
+                'Expected author_id to be str, received %s' %
+                self.author_id)
+        if not isinstance(self.placeholder, bool):
+            raise utils.ValidationError(
+                'Expected placeholder to be bool, received %s' %
+                self.placeholder)
+        if not isinstance(self.instructions, basestring):
+            raise utils.ValidationError(
+                'Expected instructions to be string,' +
+                'received %s' %self.instructions)
+
+
+class ImageAssets(object):
+    """Value object representing the images for the state."""
+    def __init__(self, image_mapping):
+        """Constructs a ImageAssets domain object.
+
+        Args:
+            image_mapping: dict. The dict representation of
+                ImageAssets mapping.
+        """
+        self.image_mapping = image_mapping
+        self.validate()
+
+    def validate(self):
+        """Validates all properties of ImageAssets.
+
+        Raises:
+            ValidationError: One or more attributes of the ImageAssets are not
+            valid.
+        """
+        for image_id in self.image_mapping:
+            image = self.image_mapping[image_id]
```

Comment:
Add validation for image_id

Reply:
One thing to say, if you see I was validating image_id at 2-3 places
1) while applying the change list. (```apply_change_list()``` function)
2) After the applied change list. (means while validating of exploration)
3) After initializing image assets.
And I covered almost all cases of image_id.
So did you think, there is a need for validation here?

Reply:
This is the main place for all the validation related to imageAssets object, other places are optional.

Reply:
done

Reply:
done

Reply:
I don't see any validation for image_id here, make sure you have addressed the comment before marking a comment resolved.

Reply:
sorry, for it. I think i have done it.
Done